### PR TITLE
codeintel: Refresh configuration policy UI

### LINKED ISSUE #46464: Refresh configuration policy UI
@jjinnii and I ran a few user studies on the UI that deals with listing and modifying code intel configuration policies a few months ago. There was a lot of obvious low-hanging fruit we should tackle to improve the UI here without any necessary backend changes or major structural changes to the UI/GraphQL queries required to resolve it.

For example: form fields only have a name and little description (maybe a hint that it's required), and uses a lot of jargon such as unqualified uses of `HEAD` and _index intermediate commits_.

### PR DESCRIPTION
**Pull request by efritz**
This PR rewrites the pages controlling code intel configuration policies. Fixes #46464.

**Changes to frontend code**:

- Deleted a bunch of unused components. New code and modified should be read basically as the new version only. Focus reading the following files:
  - `client/web/src/enterprise/codeintel/configuration/pages/CodeIntelConfigurationPage.tsx`
  - `client/web/src/enterprise/codeintel/configuration/pages/CodeIntelConfigurationPolicyPage.tsx`
  - `client/web/src/enterprise/codeintel/configuration/components/RepositoryPatternList.tsx`

**Changes to backend code (you can read these files bottom-up in the diff view)**:

- GraphQL changes:
  - Added `ExternalRepository` to the code intelligence repository resolver
  - Add limit/total count to git object preview
  - Remove pagination from repository preview
  - Added additional parameter and field to git object preview (to differentiate commits younger/older than the configured policy threshold)
- Added stronger validation of duration/age inputs
- Respect new limit/age parameter when resolving git preview
- Expose dates of commits from the policy/commit matcher to populate the new GraphQL fields

## The goodies (new screenshots)

global-list
global edit
too many repos
single repo pattern
protected
all options
tags
repo-list

## The baddies (old screenshots)

Screenshot 2023-01-19 at 3 59 07 PM
Screenshot 2023-01-19 at 3 59 13 PM
Screenshot 2023-01-19 at 3 59 37 PM

## Test plan

Tested extensively by hand.

## App preview:

- [Web](https://sg-web-ef-rewrite-config-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.00% (-0.00 kb) | -0.02% (-2.24 kb) | -0.02% (-2.24 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ef/rewrite-config-ui/compare-report.html;tab=live_object) for a full comparison between the commits 4761eb0 and a6a11c4 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: cmd/frontend/graphqlbackend/codeintel.graphql [hunk 1]
```diff
@@ -303,19 +303,9 @@ extend type Query {
...
 
         """
-        When specified, indicates that this request should be paginated and
-        the first N results (relative to the cursor) should be returned. i.e.
-        how many results to return per page.
+        When specified, indicates that this request should return the first N items.
         """
         first: Int
-
-        """
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/codeintel.graphql:312]: no more pagination?

**Review by eseliger — APPROVED** [review]
just did a pretty light review on constrained time, but the screenshots look good to me and didn't see anything standing out as critical

File: cmd/frontend/graphqlbackend/codeintel.graphql (hunk 1 above)
Comment (efritz) [reply cmd/frontend/graphqlbackend/codeintel.graphql:312]: Nope, it's only used for preview page and there's no reason given the current filtering/display that users would need to load more. Was originally overkill.

**Comment by efritz** [comment]
See the [Figma file](https://www.figma.com/file/PqRnWEXK9pJPy6T1oxw4nQ/AutoIndex-Configuration-QA?node-id=0%3A1&t=IYRVc4F3ab5idqfo-0) for design feedback on this new structure.

**Review by Numbers88s — APPROVED** [review]

**Pull request merged**